src/internal_events/nats.rs

```diff
@@ -48,16 +47,29 @@ impl InternalEvent for NatsEventSendSuccess {
 }
 
 #[derive(Debug)]
-pub struct NatsEventSendFail {
+pub struct NatsEventSendError {
     pub error: Error,
 }
 
-impl InternalEvent for NatsEventSendFail {
+impl InternalEvent for NatsEventSendError {
     fn emit_logs(&self) {
-        error!(message = "Failed to send message.", error = %self.error);
+        error!(
+            message = "Failed to send message.",
+            error = %self.error,
+            error_code = "sending_event",
```

Comment:
Maybe `send_failed`?

Reply:
Actually, I changed it for the reason mentioned and I agreed with : https://github.com/vectordotdev/vector/pull/11383#discussion_r810159463